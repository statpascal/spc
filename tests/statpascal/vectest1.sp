program vectest1;

const
    n = 1; // 100 * 1000 * 1000;
    m = 10;

var
    a, b, c: vector of real;
    d: vector of int64;
    i, retval: integer;
    s: string;

begin
    a := intvec (1, m);
    b := intvec (1, m);
    for i := 1 to n do
	c := a + b;
    writeln (c);
    writeln (sum (intvec (1, 100)));
    writeln (sum (realvec (1, 1000, 1)));
    writeln (count (intvec (1, 100) <= 50));
    writeln (cumsum (intvec (1, 20)));
    writeln (cumsum (realvec (0, 1, 0.1)));
    writeln (rev (intvec (1, 5)));
    d := intvec (1, 10);
    writeln (rev (d));
    writeln (d);
    writeln (rev (realvec (0, 1, 0.1)));
    d := randomperm (10);
    writeln (sort (d))
end.
