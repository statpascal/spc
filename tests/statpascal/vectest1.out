2 4 6 8 10 12 14 16 18 20 
5050
500500
50
1 3 6 10 15 21 28 36 45 55 66 78 91 105 120 136 153 171 190 210 
0 0.1 0.3 0.6 1 1.5 2.1 2.8 3.6 4.5 5.5 
5 4 3 2 1 
10 9 8 7 6 5 4 3 2 1 
1 2 3 4 5 6 7 8 9 10 
1 0.9 0.8 0.7 0.6 0.5 0.4 0.3 0.2 0.1 0 
1 2 3 4 5 6 7 8 9 10 
